### PR TITLE
Native edge cases

### PR DESCRIPTION
**Pull request by wcornwell**
better handling edge cases (including hyphens in species names). current behavior:

1. Warns if not binomial
2. Tries to match at the binomial level
3. Returns "unknown" for genera and/or families

### CODE REVIEW

**Comment by fontikar** [comment]
I am wondering if we should modify native_in_australia so its like update_taxonomy and accepts the a DF of aligned names??

**Comment by wcornwell** [comment]
Or a DF of `updated_names` since it won't really work with anything that's not APC accepted anyway?

**Comment by wcornwell** [comment]
current way to use the function with `create_taxonomic_update_lookup`: 
Screenshot 2023-09-07 at 12 37 20 pm

**Comment by fontikar** [comment]
> Or a DF of `updated_names` since it won't really work with anything that's not APC accepted anyway?

I really like this idea, links with not APC stuff better!!!

File: R/native_anywhere_in_australia.R [hunk 1]
```diff
@@ -1,45 +1,62 @@
...
+native_anywhere_in_australia <- function(species, resources = load_taxonomic_resources()) {
+  
+  # Remove duplicates and replace hyphens
+  species <- unique(species)
+  species_modified <- gsub('-', ' ', species)
+  
+  # Count words and issue warnings for non-binomial names
+  word_counts <- stringi::stri_count_words(species_modified)
+  if (any(word_counts != 2)) {
+    warning("All input species not binomials; \n this function is designed to work primarily with species-level binomials.")
```
Comment (fontikar) [inline R/native_anywhere_in_australia.R:29]: Replace "All" with "Some"?

**Review by fontikar — DISMISSED** [review]
Minor comment, nice fix!

**Review by wcornwell — COMMENTED** [review]
Looks great

File: R/native_anywhere_in_australia.R (hunk 1 above)
Comment (wcornwell) [reply R/native_anywhere_in_australia.R]: done

**Review by fontikar — APPROVED** [review]
LOOKS GREAT

**Pull request merged**